### PR TITLE
feat: allow `i` flag in regex patterns

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
After this change the regex patterns of the options `allowedVersions`, `baseBranches`, `matchBaseBranches`, `matcCurrentValue` , `matchCurrentVersion` and the `autodiscoverFilter` will allow the `i` flag

Logic changes have been made to the `configRegexPredicate` function, and new tests have been added to all affected functions.

## Context
https://github.com/renovatebot/renovate/issues/21286#issuecomment-1902550512

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
Test Run: https://github.com/RahulGautamSingh-testing/iflag/pulls/1

File: lib/util/regex.ts [hunk 1]
```diff
@@ -78,7 +78,8 @@ function parseConfigRegex(input: string): RegExp | null {
     const regexString = input
       .replace(configValStart, '')
       .replace(configValEnd, '');
-    return regEx(regexString);
+    const isCaseSensitive = !input.endsWith('i');
+    return isCaseSensitive ? regEx(regexString) : regEx(regexString, 'i');
```
Comment (rarkins) [inline lib/util/regex.ts:82]:
```suggestion
    return input.endsWith('i') ? regEx(regexString, 'i') : regEx(regexString));
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Any applicable docs or examples which we can update?

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
Let's solve with https://github.com/renovatebot/renovate/issues/26817

**Review by rarkins — APPROVED** [review]

**Pull request merged**